src/pytorch_lightning/loggers/wandb.py

```diff
@@ -552,6 +552,10 @@ def use_artifact(self, artifact: str, artifact_type: Optional[str] = None) -> "w
 
     @rank_zero_only
     def finalize(self, status: str) -> None:
+        if self._experiment is None:
```

Comment:
experiment is initialized during init, so will it be None on the main process?

Reply:
You're right! On wandb >= 0.12.10 that's the case. 
I changed the logic now to keep the behavior on master, which is: only upload checkpoints on success